app/models/status.rb

```diff
@@ -141,6 +143,11 @@ def non_sensitive_with_media?
     !sensitive? && media_attachments.any?
   end
 
+  before_validation :prepare_contents
+  before_create     :set_reblog
+  before_create     :set_visibility
+  before_create     :set_conversation
```

Comment:
a more rails-like way to do this is to have this hook as either `if: :local?` or `unless: :conversation_id?` instead of checking it multiple times in the method.